src/tools/rmc-link-restrictions/src/main.rs

```diff
@@ -0,0 +1,77 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0 OR MIT
+
+use rmc_restrictions::*;
+use rustc_data_structures::stable_map::FxHashMap;
+use std::env;
+use std::fs::File;
+use std::io::BufReader;
+
+fn link_function_pointer_restrictions(data_per_crate: Vec<VtableCtxResults>) -> serde_json::Value {
+    let mut output = FxHashMap::default();
+    let mut combined_possible_methods: FxHashMap<TraitDefinedMethod, Vec<std::string::String>> =
+        FxHashMap::default();
+
+    // Combine method possibilities
+    for crate_data in &data_per_crate {
+        for entry in crate_data.possible_methods.clone() {
```

Comment:
Is this the right place to put the clone?

Reply:
moved